targets/zephyr/src/zephyr_getline.c

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2016 Linaro
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <zephyr.h>
+#include <uart.h>
+#include <drivers/console/uart_console.h>
+#include "zephyr_getline.h"
+
+/* While app processes one input line, Zephyr will have another line
+   buffer to accumulate more console input. */
+static struct uart_console_input line_bufs[2];
+
+static struct nano_fifo free_queue;
+static struct nano_fifo used_queue;
+
+char *zephyr_getline(void)
+{
+	static struct uart_console_input *cmd;
+
+	/* Recycle cmd buffer returned previous time */
+	if (cmd != NULL) {
```

Comment:
As mentioned in the commit message, this file is a direct import from another project, zephyr_console_helpers, which tries to follow Zephyr codestyle (in the hope that it may be eventually merged in Zephyr tree). I can reformat it per JrS style guide, but that may only complicated future syncs from zephyr_console_helpers (these aren't expected to be frequent, and expected reason would be Zephyr API change for example).